lib/pages/admin/edit_choice_board_category.dart

```diff
@@ -0,0 +1,224 @@
+                                fit: BoxFit.cover,
+                              )),
+                    // instructions text
+                    Text(
+                      "Pick the main category image",
+                      style:
+                          TextStyle(fontSize: 32, fontWeight: FontWeight.bold),
+                    ),
+                    const SizedBox(height: 20),
+                    // buttons to take/upload images
+                    PickImageButton(
+                        label: Text("Choose from Gallery"),
+                        icon: Icon(Icons.image),
+                        onPressed: () =>
+                            pickImage(source: ImageSource.gallery)),
+                    PickImageButton(
+                        label: Text("Take a Picture"),
+                        icon: Icon(Icons.camera_alt),
+                        onPressed: () => pickImage(source: ImageSource.camera)),
+                    const SizedBox(height: 20),
+                    // field to enter the category name
+                    TextField(
+                      controller: categoryNameController,
+                      decoration: InputDecoration(
+                          hintText: widget.categoryName,
+                          border: InputBorder.none,
+                          hintStyle: TextStyle(fontWeight: FontWeight.bold)),
+                      cursorColor: Colors.white,
+                    ),
+                    const Divider(
+                      color: Colors.black38,
+                      thickness: 0.2,
+                    ),
+                    const SizedBox(height: 20),
+                    // submit to database button
+                    TextButton.icon(
+                        key: const Key("editCategoryButton"),
+                        onPressed: () => editCategory(
+                            newImage: selectedImage,
+                            newCategoryName: categoryNameController.text),
+                        icon: Icon(Icons.edit),
+                        label: const Text("Edit category"),
+                        style: const ButtonStyle(
+                            foregroundColor:
+                                MaterialStatePropertyAll(Colors.white),
+                            backgroundColor: MaterialStatePropertyAll(
+                                Color.fromARGB(255, 0, 76, 153))))
+                  ],
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  // Update category image and/or title
+  Future<void> editCategory(
+      {required File? newImage, required String? newCategoryName}) async {
+    String? imageUrl;
+    if (newCategoryName!.isEmpty && newImage == null) {
+      // If user hasn't changed either
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(content: Text("No edits made")),
+      );
+    } else if (newCategoryName.isNotEmpty && newImage != null) {
```

Comment:
I think this method would benefit from being refectored. It is very long. Can you try separating it in multiple functions?